sdk/servicebus/azservicebus/errors.go

```diff
@@ -4,3 +4,4 @@ import "errors"
 
 var ErrSenderClosed = errors.New("sender is closed and cannot be used")
 var ErrReceiverClosed = errors.New("receiver has been closed and can no longer be used")
+var ErrProcessorClosed = errors.New("processor has been closed and can no longer be used")
```

Comment:
Worth a read https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

No need to make changes now but I think this is something we'll want to revisit.

Reply:
Hm....that's an interesting one.

Thinking about this a bit more, the real reason these errors exist is because you've either abruptly closed the connection/link and were in the middle of an operation _or_ you closed something and tried to use it afterwards.

In both cases the link/processor isn't coming back. So maybe I could roll all these up and just say it's ErrClosed or similar. Because I think the default thought for most errors coming back from the clients is "I should retry this".

Reply:
Hm....that's an interesting one.

Thinking about this a bit more, the real reason these errors exist is because you've either abruptly closed the connection/link and were in the middle of an operation _or_ you closed something and tried to use it afterwards.

In both cases the link/processor isn't coming back. So maybe I could roll all these up and just say it's ErrClosed or even just having an error type that's basically "FatalError" or similar? 

Because I think the default thought for most errors coming back from the clients is "I should retry this" so I just want to signal that somehow.

Reply:
As an example, for non-transient errors we've been using an interface-based pattern like [`NonRetriable()`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/internal/errorinfo/errorinfo.go#L15).  This way, it doesn't matter what type of error is returned, if it implements the `NonRetriable` interface then we know it's non-transient.

I suspect that we'll want to move away from sentinel error types and use an interface-based approach.  This way the underlying error type doesn't matter and they can be extended over time.

Reply:
Let me swap over now - there's no harm in it and it's better :)

Reply:
Okay, latest commit does what I believe you were saying. Let me know if I got it.

Reply:
(Looks like we're good on this. Resolving)